router/batchrouter/batchrouter.go

```diff
@@ -246,13 +247,20 @@ func (brt *HandleT) postToWarehouse(batchJobs BatchJobsT, location string) (err
 	jsonPayload, err := json.Marshal(&payload)
 
 	uri := fmt.Sprintf(`%s/v1/process`, warehouseURL)
-	_, err = brt.netHandle.Post(uri, "application/json; charset=utf-8",
-		bytes.NewBuffer(jsonPayload))
-	if err != nil {
-		logger.Errorf("BRT: Failed to route staging file URL to warehouse service@%v, error:%v", uri, err)
-	} else {
-		logger.Infof("BRT: Routed successfully staging file URL to warehouse service@%v", uri)
+	backoff := 0
+	for i := 0; i < noOfRetriesToWarehouseService; i++ {
+		_, err = brt.netHandle.Post(uri, "application/json; charset=utf-8",
+			bytes.NewBuffer(jsonPayload))
+		if err != nil {
+			logger.Errorf("BRT: Failed to route staging file URL to warehouse service@%v, error:%v", uri, err)
+			backoff = 2*backoff + 1 // what if back off exceeds its data type capacity
+			time.Sleep(time.Duration(backoff) * time.Second)
```

Comment:
@gane5hvarma 
let's have a maximum_backoff time like 1 min or so. Instead of `noOfRetriesToWarehouseService` let us take the time before aborting a job and abort the job after it has been trying for that much time. Check and remove the non-abort logic for warehouse jobs in `setJobStatus` func.

Also, should we do this backoff synchronously? One of the workers will remain idle due to this. It might be a problem in our hosted environment if say `noOfWorkers` destinations have wrong config / throwing errors etc.

Reply:
This is the same problem we have been discussing with router.

Should the worker sleep altogether if something is wrong instead of the job getting delayed?